sparse/rest/handlers.go

```diff
@@ -22,6 +22,26 @@ type SyncFileStub struct{}
 
 func (f *SyncFileStub) UpdateSyncFileProgress(size int64) {}
 
+func (server *SyncServer) getQueryDirectIO(request *http.Request) (bool, error) {
+	var directIO bool
+	var err error
+
+	queryParams := request.URL.Query()
+	directIOStr := queryParams.Get("directIO") // only one value for key directIO
+	if directIOStr == "" {
+		err = fmt.Errorf("queryParams directIO does not exist")
+		return directIO, err
+	}
+
+	directIO, err = strconv.ParseBool(directIOStr)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseBool(directIOStr) error: %s", err)
+		return directIO, err
+	}
+
+	return directIO, err
```

Comment:
I would prefer this kind of style:
```suggest
	queryParams := request.URL.Query()
	directIOStr := queryParams.Get("directIO")
	if directIOStr == "" {
		return false, fmt.Errorf("directIO does not exist in the queries: %+v", queryParams)
	}

	directIO, err := strconv.ParseBool(directIOStr)
	if err != nil {
		err = fmt.Errorf("strconv.ParseBool(directIOStr) error: %s", err)
		return false, errors.Warpf(err, "failed to parse directIO %v", directIOStr)
	}
	return directIO, nil
```

Returning `return directIO, err` in the last line is a little bit confusing.

Reply:
updated